carapace-server/src/test/java/org/carapaceproxy/server/mapper/BasicStandardEndpointMapperTest.java

```diff
@@ -439,4 +427,101 @@ public void testCustomAndDebbugingHeaders() throws Exception {
             }
         }
     }
+
+    @Test
+    public void testActionRedirect() throws Exception {
+
+        try (HttpProxyServer server = new HttpProxyServer(null, tmpDir.newFolder())) {
+            Properties configuration = new Properties();
+            configuration.put("listener.1.host", "0.0.0.0");
+            configuration.put("listener.1.port", "1425");
+            configuration.put("listener.1.ssl", "false");
+            configuration.put("listener.1.enabled", "true");
+
+            // redirect to same domain/uri but with https
+            configuration.put("route.1.id", "r1");
+            configuration.put("route.1.enabled", "true");
+            configuration.put("route.1.match", "regexp .*index\\.html");
+            configuration.put("route.1.action", "a1");
+            configuration.put("action.1.id", "a1");
+            configuration.put("action.1.enabled", "true");
+            configuration.put("action.1.type", "redirect");
+            configuration.put("action.1.code", "301");
+            configuration.put("action.1.redirect.proto", "https");
```

Comment:
I am expecting here that the client will receive a Location header like:
https://HOST-HEADER/same-original-path

where HOST-HEADER is exactly the value of the "Host" HTTP Header inside the Request